### ISSUE TITLE
[Regression] Default audio track is chosen when built-in audio switcher is enabled

### ISSUE THREAD

**Issue by nyakze** [clsid2/mpc-hc#2673]
**Reproduction:**
1. Install MPC-HC Version 2.1.7.2;
2. Set audio track language priorities in LAV Splitter (should differ from default audio track);
3. Enable internal audio switcher;
4. Load a file with multiple language audio tracks (tested on both local and remote).

**What should happen:**
Audio track of the selected (in LAV Splitter) language should load.

**What actually happens:**
Default audio track is loaded, ignoring priorities in LAV Splitter.

**Workarounds:**
Disable built-in audio switcher.

**Did it work before:**
MPC-HC Version 2.1.6.8 does not have this issue. Built-in audio switcher doesn't interfere with LAV Splitter audio track language priorities.

**Comment by clsid2** [comment]
It has option to remember selected track. If you select correct one, and then close and re-open, does it still select wrong one?

**Comment by nyakze** [comment]
> If you select correct one, and then close and re-open, does it still select wrong one?

You mean "RememberTrackSelection" in the advanced option? It is set to True (default). Changing the audio track and reopening the file opens the correct one (the one I selected).

**Comment by clsid2** [comment]
Then wrong track was probably stored in history before you changed splitter settings.

It correctly selects preferred language here for a file that isn't in history yet.

**Comment by nyakze** [comment]
> Then wrong track was probably stored in history before you changed splitter settings.
> 
> It correctly selects preferred language here for a file that isn't in history yet.

I have been careful to try it only with new files. It started happening on 2.1.7. I have then installed 2.1.6 and repeated the process with the new files that I've never opened.

**Comment by clsid2** [comment]
Give link to a file where it fails.
And give your exact language settings you have in LAV Splitter.

**Comment by nyakze** [comment]
> Give link to a file where it fails. 

I don't think that I am allowed to post it on a public resource. I had this behavior with all of the files which didn't have English as default. The ones I tried all had Russian audio as default and English as track 2

>And give your exact language settings you have in LAV Splitter.

In audio:
"jpn,spa,eng"

**Comment by nyakze** [comment]
The issue is non-existent if I disable built-in audio switcher. I'll just keep it disabled as I don't really need it at the moment.

**Comment by clsid2** [comment]
Audio Switcher is enabled here.

You can give torrent link here. I can remove your message afterwards.

**Comment by nyakze** [comment]
> Audio Switcher is enabled here.
> 
> You can give torrent link here. I can remove your message afterwards.

Ok, here's the ones I've tried:
magnet:?xt=urn:btih:4C3B4AF62B1963998661456B564E01B882FB2B64
magnet:?xt=urn:btih:46E3DD89DFB2707A6A9978121BE2CA9E957C0C25
magnet:?xt=urn:btih:2DC2F2C68D70ADF347AC24A1766C5BB34C590E47
magnet:?xt=urn:btih:CAF36855457A3CA63267F83822E54F0D7DE5192F
Via TorrServer and locally, result is the same.

**Comment by nyakze** [comment]
I'm not sure why it happens for me, but switching back to a previous version or disabling audio-switcher does for some reason fix this.

**Comment by clsid2** [comment]
MPC-HC options > Playback >Default Track Preference

This value may be the cause. Empty that box.

**Comment by nyakze** [comment]
> MPC-HC options > Playback >Default Track Preference
> 
> This value may be the cause. Empty that box.

It was empty. I have tried different things and after each try I restarted a player and loaded a new episode that is unknown to MPC-HC.
Reminder: trying to load eng track automatically. 
type in nonsense - rus (1) track is loaded
remove it - rus track is loaded
type in "eng" - eng track is loaded (allow overriding external splitter choice is disabled!)

Looks like the option is either overriding LAV Splitter value no matter what, or MPC doesn't see it. I tried changing audio priority in LAV Splitter from "jap,spa,eng" to just "eng", no change.
